### ISSUE TITLE
Cannot unsqueeze empty tensor

### ISSUE THREAD

**Issue by ksurya** [allenai/allennlp#2660]
**System (please complete the following information):**
 - OS: OSX
 - Python version: 3.6.5
 - AllenNLP version: 0.8.1
 - PyTorch version: 0.4.1

**Question**
How can I have empty `fields.ListField`? 

I am having `fields.ListField([ text_field.empty_field() ])` as suggested in https://github.com/allenai/allennlp/issues/1391

but Torch raises `cannot unsqueeze empty tensor` on executing `instance.as_tensor_dict()` or `iterator(..)`

![image](https://user-images.githubusercontent.com/932927/55182217-95e5fa80-5163-11e9-8d50-3de360c58968.png)

**Comment by brendan-ai2** [comment]
Here's a repro:

```
import torch
from allennlp.data.token_indexers.single_id_token_indexer import SingleIdTokenIndexer
token_indexers = {"tokens": SingleIdTokenIndexer()}
from allennlp.data.tokenizers.word_tokenizer import WordTokenizer
tokenizer = WordTokenizer()
tokens = tokenizer.tokenize("Foo")
from allennlp.data.fields.text_field import TextField
from allennlp.data.fields.list_field import ListField
text_field = TextField(tokens, token_indexers)
list_field = ListField([text_field.empty_field()])
fields = {'list': list_field}
from allennlp.data.instance import Instance
instance = Instance(fields)
instance.as_tensor_dict()
```

**Comment by brendan-ai2** [comment]
I think this bug is due to there not be any non-empty lists in your batch. If there were, the aggregated padding would prevent this bug. I have a speculative fix in https://github.com/allenai/allennlp/pull/2697, but I'm really not sure if it's the right approach.

**Comment by DeNeutoy** [comment]
Hi @ksurya - could you let us know a bit more about how you are using the `ListField(textfield.empty_field())`? We've found that in many cases, you might be wanting to model something in a way which might be easier if approached differently, but equally we don't want allennlp to crash in this scenario either. It would be helpful to see some code, (maybe your `DatasetReader`) and hear a bit about the model you are trying to build. Thanks!

**Comment by ksurya** [comment]
Hi guys, appreciate your interest on this issue. I will get back with a Jupyter Notebook reproducing this problem and let you know.
